### DISCUSSION TITLE
Please suport unicode-range in the Font-face block

### DISCUSSION THREAD

**Discussion by Robert-Babiak** [dompdf/dompdf#3516] [Ideas]
Hello.

We have a use case where we have an HYTML page that containse both Chinese and Cree text. In chrome we load both fonts and it renders correctly, when converting to PDF it chooses the first font and uses that so either the Chinese or Cree is not rendered depending on the order of the fonts listed in the Font-family CSS. 

Thank you for considering my request.

**Comment by bsweeney** [comment]
We have an open enhancement request for this, follow #913.

That being said, which release of Dompdf are you using? Version 3.0.0 added glyph parsing support so that if one font doesn't support a character Dompdf will check the fallback.

You don't supply a sample document, but using your scenario as an example the following HTML:

```
<!DOCTYPE html>
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
  <link rel="preconnect" href="https://fonts.googleapis.com">
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
  <link href="https://fonts.googleapis.com/css2?family=Noto+Sans+Canadian+Aboriginal&amp;family=Noto+Sans+TC&amp;display=swap" rel="stylesheet">
  <style>
    * {
      font-family: "Noto Sans TC", "Noto Sans Canadian Aboriginal", "DejaVu Sans", sans-serif;
    }
  </style>
</head>
<body>
  <p>Chinese: 鑑於對人類家庭所有成員的固有尊嚴及其平等的和不移的權利的承認，乃是世界自由、正義與和平的基礎</p>
  <p>Cree: ᐃᒪᐃᒻᒪᑦ ᐃᓕᑕᖅᓯᒪᐅᑎᖃᑦᒪᑦ ᓯᕗᓕᕐᓂᓴᑐᖃᕐᓂᒃ ᓂᕐᓱᐃᓂᑦᒥᒃ</p>
</body>
</html>
```

the output looks like this in Dompdf <3.0.0
![image](https://github.com/user-attachments/assets/95483d7f-0cb3-4fa2-b33e-1841cb1a7a1b)

but looks like this in Dompdf >= 3.0.0
![image](https://github.com/user-attachments/assets/f6e41c2e-e098-4fff-b287-5c41cd43038b)

While this is not true Unicode range support, so long as your fonts have a correct glyph index you should get the output you want.

**Comment by Robert-Babiak** [comment]
thank you, I will track that and verify what version we have installed, if
we are not on three I will investigate updating to it.

Thank you for the prompt response.
